StrongInject.Generator/ContainerGenerator.cs

```diff
@@ -316,6 +316,9 @@ private DisposalLowerer CreateDisposalLowerer(bool disposeAsynchronously)
         {
             _cancellationToken.ThrowIfCancellationRequested();
 
+            var endingNonThrowingOperationCount = operations.Reverse().TakeWhile(operation => operation.Statement is OwnedCreationStatement).Count();
```

Comment:
I isolated the reason in the last commit so that you can see the diff in the test. It's a nice-to-have.

Maybe I'll unsquash the previous commit so that the test diff explains earlier steps, too.